tests/Builder/Accumulator/DenseRankAccumulatorTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Accumulator;
+
+use MongoDB\Builder\Accumulator;
+use MongoDB\Builder\Expression;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+use function MongoDB\object;
+
+/**
+ * Test $denseRank accumulator
+ */
+class DenseRankAccumulatorTest extends PipelineTestCase
+{
+    public function testDenseRankPartitionsByADateField(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::setWindowFields(
+                partitionBy: Expression::stringFieldPath('state'),
+                sortBy: object(
+                    orderDate: 1,
+                ),
+                output: object(
+                    denseRankOrderDateForState: Accumulator::outputWindow(
+                        Accumulator::denseRank(),
+                    ),
```

Comment:
The `outputWindow` is optional when no window property is set. I omitted it in the next test.

Reply:
Added this note as a comment in the code.